Content.Server/Objectives/Systems/StealConditionSystem.cs

```diff
@@ -44,11 +47,45 @@ public override void Initialize()
     /// start checks of target acceptability, and generation of start values.
     private void OnAssigned(Entity<StealConditionComponent> condition, ref ObjectiveAssignedEvent args)
     {
+        // Select map based on the filter mode
+        var mapFilterMode = condition.Comp.MapFilterMode;
+        MapId selectedMapId = default;
+        var hasSelectedMap = false;
+
+        switch (mapFilterMode)
+        {
+            case StealMapFilterMode.AnyMap:
+                // No map filtering
+                hasSelectedMap = false;
```

Comment:
This is already set to false.

I think what I would do is change this to a switch expression. Something like this:
```cs
MapId? selectedMapId = condition.Comp.MapFilterMode switch
{
    StealMapFilterMode.AnyMap => null,
    StealMapFilterMode.DefaultMap => _gameTicker.DefaultMap,
    StealMapFilterMode.MindOwnerCurrentMap when args.Mind.OwnedEntity is EntityUid owner => Transform(owner).MapID,
    // Fallback in case the mind isn't attached to a body
    StealMapFilterMode.MindOwnerCurrentMap => _gameTicker.DefaultMap,
}
```
Much shorter and more readable that way.